tools/target-server-validator/callout/src/main/java/com/google/apigee/json/JavaxJson.java

```diff
@@ -0,0 +1,190 @@
+// JavaxJson.java
+
+package com.google.apigee.json;
+
+import java.beans.Introspector;
+import java.beans.PropertyDescriptor;
+import java.io.StringReader;
+import java.lang.reflect.Array;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonNumber;
+import javax.json.JsonObject;
+import javax.json.JsonString;
+import javax.json.JsonValue;
+import javax.json.JsonValue.ValueType;
+
+public class JavaxJson {
+  @SuppressWarnings("unchecked")
+  public static <T> T fromJson(String json, Class<T> beanClass) {
+    JsonValue value = Json.createReader(new StringReader(json)).read();
+    return (T) decode(value, beanClass);
+  }
+
+  private static Object decode(JsonValue jsonValue, Type targetType) {
```

Comment:
What's the rationale for mixing getValueType Enums and instanceof? Would it be more readable to just use the enum and turn it into a switch?